### PR TITLE
Add Frameworks Samples

### PR DESCRIPTION
**Pull request by averikitsch**

### CODE REVIEW

File: appengine-java11/kotlin-ktor/pom.xml [hunk 1]
```diff
@@ -0,0 +1,162 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (lesv) [inline appengine-java11/kotlin-ktor/pom.xml]: With kotlin you might be better off using gradle, but it's ok here.

File: appengine-java11/kotlin-ktor/pom.xml [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+        <kotlin.code.style>official</kotlin.code.style>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+        <kotlin.compiler.incremental>true</kotlin.compiler.incremental>
+        <main.class>io.ktor.server.netty.EngineMain</main.class>
+        <maven.compiler.target>11</maven.compiler.target>
+        <maven.compiler.source>11</maven.compiler.source>
+    </properties>
+
+    <repositories>
+        <repository>
```
Comment (lesv) [inline appengine-java11/kotlin-ktor/pom.xml]: Do you really need to use these repos?

If your going to use an alternate repo, consider ours: http://maven-central.storage.googleapis.com/index.html
(unless we don't have the artifacts your looking for)

File: appengine-java11/kotlin-ktor/pom.xml [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+            <snapshots><enabled>true</enabled></snapshots>
+        </repository>
+        <repository>
+            <id>repo2</id>
+            <url>https://kotlin.bintray.com/ktor</url>
+            <releases><enabled>true</enabled></releases>
+            <snapshots><enabled>true</enabled></snapshots>
+        </repository>
+    </repositories>
+    <dependencies>
```
Comment (lesv) [inline appengine-java11/kotlin-ktor/pom.xml]: whitespace makes it easier to read

File: appengine-java11/kotlin-ktor/pom.xml (hunk 1 above)
Comment (lesv) [inline appengine-java11/kotlin-ktor/pom.xml]: License & Copyright?

File: appengine-java11/kotlin-ktor/pom.xml [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+            <version>1.1.3</version>
+        </dependency>
+        <dependency>
+            <groupId>io.ktor</groupId>
+            <artifactId>ktor-server-tests</artifactId>
+            <version>1.1.3</version>
+            <scope>test</scope>
+        </dependency>
+    </dependencies>
+    <build>
```
Comment (lesv) [inline appengine-java11/kotlin-ktor/pom.xml]: whitespace?

File: appengine-java11/kotlin-ktor/pom.xml [hunk 5]
```diff
@@ -0,0 +1,162 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
```
Comment (lesv) [inline appengine-java11/kotlin-ktor/pom.xml]: Feel free to ignore - I appear to be a bit pedantic today.
should be 2 spaces, not 4
https://google.github.io/styleguide/xmlstyle.html

File: appengine-java11/kotlin-ktor/pom.xml [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+    </dependencies>
+    <build>
+        <sourceDirectory>${project.basedir}/src</sourceDirectory>
+        <testSourceDirectory>${project.basedir}/test</testSourceDirectory>
+        <resources>
+            <resource>
+                <directory>${project.basedir}/resources</directory>
+            </resource>
+        </resources>
+        <plugins>
```
Comment (lesv) [inline appengine-java11/kotlin-ktor/pom.xml]: There is a lot of stuff in here -- perhaps a comment or two might help a developer figure out what's happening.

File: appengine-java11/kotlin-ktor/resources/application.conf [hunk 7]
```diff
@@ -0,0 +1,11 @@
+ktor {
```
Comment (lesv) [inline appengine-java11/kotlin-ktor/resources/application.conf]: Copyright & License?

File: appengine-java11/kotlin-ktor/resources/logback.xml [hunk 8]
```diff
@@ -0,0 +1,12 @@
+<configuration>
```
Comment (lesv) [inline appengine-java11/kotlin-ktor/resources/logback.xml]: Copyright & License

File: appengine-java11/kotlin-ktor/src/Application.kt [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+import io.ktor.http.content.*
+import io.ktor.server.engine.*
+
+fun main(args: Array<String>): Unit = io.ktor.server.netty.EngineMain.main(args)
+
+@Suppress("unused") // Referenced in application.conf
+@kotlin.jvm.JvmOverloads
+fun Application.module(testing: Boolean = false) {
+    install(ShutDownUrl.ApplicationCallFeature) {
+        // The URL that will be intercepted (you can also use the application.conf's ktor.deployment.shutdown.url key)
```
Comment (lesv) [inline appengine-java11/kotlin-ktor/src/Application.kt]: line length?

File: appengine-java11/micronaut-helloworld/pom.xml [hunk 10]
```diff
@@ -0,0 +1,211 @@
...
+    <dependency>
+      <groupId>io.micronaut.test</groupId>
+      <artifactId>micronaut-test-junit5</artifactId>
+      <scope>test</scope>
+    </dependency>
+  </dependencies>
+
+  <build>
+    <plugins>
+      <plugin>
```
Comment (lesv) [inline appengine-java11/micronaut-helloworld/pom.xml]: Some comments might be in order here

File: appengine-java11/micronaut-helloworld/src/main/resources/logback.xml [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+-->
+
+<configuration>
+
+    <appender name="STDOUT" class="ch.qos.logback.core.ConsoleAppender">
+        <withJansi>true</withJansi>
+        <!-- encoders are assigned the type
+             ch.qos.logback.classic.encoder.PatternLayoutEncoder by default -->
+        <encoder>
+            <pattern>%cyan(%d{HH:mm:ss.SSS}) %gray([%thread]) %highlight(%-5level) %magenta(%logger{36}) - %msg%n</pattern>
```
Comment (lesv) [inline appengine-java11/micronaut-helloworld/src/main/resources/logback.xml]: Long line -- may not be anything you can do about it.

File: appengine-java11/quarkus-helloworld/pom.xml [hunk 12]
```diff
@@ -0,0 +1,94 @@
+<?xml version="1.0"?>
+<project xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd" xmlns="http://maven.apache.org/POM/4.0.0"
```
Comment (lesv) [inline appengine-java11/quarkus-helloworld/pom.xml]: very long line

File: appengine-java11/quarkus-helloworld/pom.xml [hunk 13]
```diff
@@ -0,0 +1,94 @@
...
+  <modelVersion>4.0.0</modelVersion>
+  <groupId>com.example.appengine.quarkus</groupId>
+  <artifactId>quarkus-helloworld</artifactId>
+  <version>1.0-SNAPSHOT</version>
+
+  <!--
+  The parent pom defines common style checks and testing strategies for our samples.
+  Removing or replacing it should not affect the execution of the samples in anyway.
+  -->
+  <!-- <parent>
```
Comment (lesv) [inline appengine-java11/quarkus-helloworld/pom.xml]: Should this be commented out?

File: appengine-java11/quarkus-helloworld/pom.xml [hunk 14]
```diff
@@ -0,0 +1,94 @@
...
+        <configuration>
+          <systemProperties>
+            <java.util.logging.manager>org.jboss.logmanager.LogManager</java.util.logging.manager>
+          </systemProperties>
+        </configuration>
+      </plugin>
+      <plugin>
+        <groupId>com.google.cloud.tools</groupId>
+        <artifactId>appengine-maven-plugin</artifactId>
+        <version>2.0.0-rc5</version>
```
Comment (lesv) [inline appengine-java11/quarkus-helloworld/pom.xml]: they've released now. s.b. 2.0.0

File: appengine-java11/quarkus-helloworld/src/main/appengine/app.yaml [hunk 15]
```diff
@@ -0,0 +1,19 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+runtime: java11
+instance_class: F1
+# The entrypoint here is mandatory as the appengine-staging area contains an
+# exploded fatjar.
+entrypoint: 'java -XX:MaxRAM=128m -XX:ActiveProcessorCount=2 -Xmx32m -Dquarkus.log.file.enable=false -Dquarkus.http.host=0.0.0.0 -jar quarkus-helloworld-1.0-SNAPSHOT-runner.jar'
```
Comment (lesv) [inline appengine-java11/quarkus-helloworld/src/main/appengine/app.yaml]: Do you get two processors in GAE?  I've recently heard yes, but do you know that is true?

File: appengine-java11/quarkus-helloworld/src/main/resources/META-INF/microprofile-config.properties [hunk 16]
```diff
@@ -0,0 +1 @@
+# Configuration file
```
Comment (lesv) [inline appengine-java11/quarkus-helloworld/src/main/resources/META-INF/microprofile-config.properties]: I'm sure this is required, perhaps w/ copyright & license ?

File: appengine-java11/sparkjava-helloworld/pom.xml [hunk 17]
```diff
@@ -0,0 +1,112 @@
...
+  <modelVersion>4.0.0</modelVersion>
+  <groupId>com.example.appengine.sparkdemo</groupId>
+  <artifactId>sparkjava-helloworld</artifactId>
+  <version>1.0</version>
+
+  <!--
+  The parent pom defines common style checks and testing strategies for our samples.
+  Removing or replacing it should not affect the execution of the samples in anyway.
+  -->
+  <!-- <parent>
```
Comment (lesv) [inline appengine-java11/sparkjava-helloworld/pom.xml]: commented out?

File: appengine-java11/sparkjava-helloworld/src/main/appengine/app.yaml [hunk 18]
```diff
@@ -0,0 +1,2 @@
+runtime: java11
```
Comment (lesv) [inline appengine-java11/sparkjava-helloworld/src/main/appengine/app.yaml]: Copyright & License?

File: appengine-java11/sparkjava-helloworld/pom.xml [hunk 19]
```diff
@@ -0,0 +1,112 @@
...
+          <arguments>
+            <argument>-jar</argument>
+            <argument>${app.stage.stagingDirectory}/sparkjava-helloworld-1.0-jar-with-dependencies.jar</argument>
+          </arguments>
+        </configuration>
+      </plugin>
+      <plugin>
+        <groupId>com.google.cloud.tools</groupId>
+        <artifactId>appengine-maven-plugin</artifactId>
+        <version>2.0.0-rc5</version>
```
Comment (lesv) [inline appengine-java11/sparkjava-helloworld/pom.xml]: 2.0.0

File: appengine-java11/vertx-helloworld/pom.xml [hunk 20]
```diff
@@ -0,0 +1,108 @@
...
+            <manifest>
+              <mainClass>com.example.appengine.vertxhello.Main</mainClass>
+            </manifest>
+          </archive>
+        </configuration>
+      </plugin>
+      <plugin>
+        <groupId>com.google.cloud.tools</groupId>
+        <artifactId>appengine-maven-plugin</artifactId>
+        <version>2.0.0-rc5</version>
```
Comment (lesv) [inline appengine-java11/vertx-helloworld/pom.xml]: version

File: appengine-java11/kotlin-ktor/pom.xml (hunk 2 above)
Comment (averikitsch) [reply appengine-java11/kotlin-ktor/pom.xml]: These are the most up to date repos. Ktor doesn't keep their mvn central repo as updated.

**Comment by averikitsch** [comment]
Tests are passing now. Styling+licenses have been updated.

**Pull request merged**